EntechTestService.Services.Validation.UnitTests/StoreNameValidatorTest.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using EntechTestService.Contracts.Internal.Model;
+using EntechTestService.Contracts.Internal.Repositories;
+
+namespace EntechTestService.Services.Validation.UnitTests
+{
+    [TestClass]
+    public class StoreNameValidatorTest
+    {
+        private class FakeRepository : IStoreRepository
+        {
+            private readonly string duplicateName;
+
+            private readonly int idOfDuplicateStore;
+
+            public FakeRepository(string duplicateName, int idOfDuplicateStore)
+            {
+                this.duplicateName = duplicateName;
+                this.idOfDuplicateStore = idOfDuplicateStore;
+            }
+
+            public int CreateStore(StoreData newProductData)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void DeleteStore(int id)
+            {
+                throw new NotImplementedException();
+            }
+
+            public ICollection<IdentifiedDataEntity<StoreData>> FindStores(Expression<Predicate<StoreData>> filter)
+            {
+                if (filter.Compile()(new StoreData { Name = duplicateName }))
+                {
+                    return new[] { new IdentifiedDataEntity<StoreData>(idOfDuplicateStore, new StoreData { Name = duplicateName }) };
+                }
+
+                return new IdentifiedDataEntity<StoreData>[0];
+            }
+
+            public StoreData GetStore(int id)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void UpdateStore(int id, StoreData newProductData)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private readonly Random random;
```

Comment:
Unfortunately, I do not have a time to redesign the tests today.
I've replaced random with deterministic numbers generator (so that every test run will use the same numbers).